### PR TITLE
chore: add devfile v2, use Code - OSS

### PR DESCRIPTION
**Pull request by themr0c**
## What does this pull request change

* chore: add devfilev2
* chore: use Visual Studio Code
* chore: add theia settings

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: devfilev2.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        - "https://github.com/golang/vscode-go/releases/download/v0.23.0/go-0.23.0.vsix"
+        - "https://github.com/microsoft/vscode-python/releases/download/2020.7.94776/ms-python-release.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/7.44-che-assets/java-0.82.0-369.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/v069c634/vscode-xml-0.17.0.vsix"
+        - "https://open-vsx.org/api/redhat/vscode-yaml/0.14.0/file/redhat.vscode-yaml-0.14.0.vsix"
+        - "https://open-vsx.org/api/vscode/typescript-language-features/1.49.3/file/vscode.typescript-language-features-1.49.3.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/7.44-che-assets/vscode-java-debug-0.26.0.vsix"
+        - "https://open-vsx.org/api/vscjava/vscode-java-test/0.28.1/file/vscjava.vscode-java-test-0.28.1.vsix"
+      che-theia.eclipse.org/vscode-preferences:
+        java.server.launchMode: Standard
```
Comment (themr0c) [inline devfilev2.yaml:17]: Is it possible to remove these from the devfile and add to a specific configuration file?

File: devfilev2.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
+          value: /remote-endpoint/plugin-remote-endpoint
+        - name: THEIA_PLUGINS
+          value: "local-dir:///plugins/sidecars/che-docs"
+        - name: CHE_DASHBOARD_URL
+          value: "https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com"
+        - name: CHE_PLUGIN_REGISTRY_URL
+          value: "https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/plugin-registry/v3"
+        - name: CHE_PLUGIN_REGISTRY_INTERNAL_URL
+          value: "http://plugin-registry.dogfooding.svc:8080/v3"
```
Comment (themr0c) [inline devfilev2.yaml]: Is it possible or advised to remove all these env variables?

```suggestion
```

File: devfilev2.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        - "https://github.com/microsoft/vscode-python/releases/download/2020.7.94776/ms-python-release.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/7.44-che-assets/java-0.82.0-369.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/v069c634/vscode-xml-0.17.0.vsix"
+        - "https://open-vsx.org/api/redhat/vscode-yaml/0.14.0/file/redhat.vscode-yaml-0.14.0.vsix"
+        - "https://open-vsx.org/api/vscode/typescript-language-features/1.49.3/file/vscode.typescript-language-features-1.49.3.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/7.44-che-assets/vscode-java-debug-0.26.0.vsix"
+        - "https://open-vsx.org/api/vscjava/vscode-java-test/0.28.1/file/vscjava.vscode-java-test-0.28.1.vsix"
+      che-theia.eclipse.org/vscode-preferences:
+        java.server.launchMode: Standard
+    container:
```
Comment (themr0c) [inline devfilev2.yaml]: Is it possible to move these attributes to another configuration file?

```suggestion
  - container:
```

File: devfilev2.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      sourceMapping: /projects
+      volumeMounts:
+        - name: venv
+          path: /home/theia/.venv
+        - name: m2
+          path: /home/theia/.m2
+        - name: remote-endpoint
+          path: /remote-endpoint
+        - name: plugins
+          path: /plugins
```
Comment (themr0c) [inline devfilev2.yaml]: Is it required to specify these volumeMounts?
```suggestion
```

File: devfilev2.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    name: che-docs
+  - name: venv
+    volume: {}
+  - name: m2
+    volume: {}
```
Comment (themr0c) [inline devfilev2.yaml]: Are these volumes required?

```suggestion
```

File: devfilev2.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+components:
+  - attributes:
+      app.kubernetes.io/name: che-docs
+      che-theia.eclipse.org/vscode-extensions:
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/v069c634/vscode-xml-0.17.0.vsix"
+        - "https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/vf707df2/shellcheck-v0.13.2.vsix"
+        - "https://open-vsx.org/api/redhat/vscode-yaml/0.14.0/file/redhat.vscode-yaml-0.14.0.vsix"
+      che-theia.eclipse.org/vscode-preferences:
+        shellcheck.executablePath: /bin/shellcheck
+    container:
```
Comment (themr0c) [inline devfilev2.yaml:13]: > Danger alert: Container che-docs has state CrashLoopBackOff

On:

Dashboard Version
    7.50.0-next
Server Version
    7.48.0-SNAPSHOT

File: devfilev2.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+#
```
Comment (themr0c) [inline devfilev2.yaml:1]: Still not entirely operational: 
> devworkspace failed to progress past phase 'Starting' for longer than timeout (5m)

**Review by amisevsk — COMMENTED** [review]
See also: https://github.com/eclipse-che/che-docs/pull/2229 (merged-into branch has since been deleted)

File: devfilev2.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+schemaVersion: 2.1.0
+metadata:
+  name: che-docs
+attributes:
+  controller.devfile.io/storage-type: ephemeral
```
Comment (themr0c) [inline devfilev2.yaml:13]: The ephemeral option is a very dangerous one for contributors. You loose all your contributions if for any reason you cannot commit and push before the workspace stop. Are you sure we want this? I find the `generateName` option, creating a new workspace when you don't restart an existing workspace less dangerous.

File: devfilev2.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+# Copyright (c) 2022 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+schemaVersion: 2.1.0
+metadata:
+  name: che-docs
```
Comment (themr0c) [inline devfilev2.yaml:11]: I prefer generateName for that reason:

> Danger alert:Failed to create a workspace. Failed to create a new workspace from the devfile, reason: Failed to create a new workspace. Unable to create devworkspace: devworkspaces.workspace.devfile.io "che-docs" already exists

**Review by l0rd — APPROVED** [review]

**Pull request merged**